adam-core/src/main/scala/org/bdgenomics/adam/rdd/contig/NucleotideContigFragmentRDD.scala

```diff
@@ -392,6 +386,46 @@ sealed abstract class NucleotideContigFragmentRDD extends AvroGenomicRDD[Nucleot
   }
 
   /**
+   * From a set of contigs, returns list of sequences based on reference regions provided
+   * @param regionsList List of Reference regions over which to get sequences
+   * @return RDD[(ReferenceRegion, String)] of region -> sequence pairs
+   */
+  def extractRegions(regionsList: List[ReferenceRegion]): RDD[(ReferenceRegion, String)] = {
+
+    val regions = regionsList.distinct //to avoid duplicates
+
+    def extractSequence(fragmentRegion: ReferenceRegion, fragment: NucleotideContigFragment, region: ReferenceRegion): (ReferenceRegion, String) = {
+      val merged = fragmentRegion.intersection(region)
+      val start = (merged.start - fragmentRegion.start).toInt
+      val end = (merged.end - fragmentRegion.start).toInt
+      val fragmentSequence: String = fragment.getSequence
+      (merged, fragmentSequence.substring(start, end))
+    }
+
+    def reduceRegionSequences(
+      kv1: (ReferenceRegion, String),
+      kv2: (ReferenceRegion, String)): (ReferenceRegion, String) = {
+      (kv1._1.merge(kv2._1), if (kv1._1.compareTo(kv2._1) <= 0) {
+        kv1._2 + kv2._2
+      } else {
+        kv2._2 + kv1._2
+      })
+    }
+
+    val byRegion: RDD[(Option[ReferenceRegion], NucleotideContigFragment)] = rdd.keyBy(ReferenceRegion(_))
```

Comment:
Thanks, I did not know about this function (probably because it's protected and I am porting the method from my adam extensions lib)...